spec/jobs/part_replication_audit_job_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+describe PartReplicationAuditJob, type: :job do
+  let(:cm) { create(:complete_moab, version: 2) }
+  let(:job) { described_class.new(cm, endpoint) }
+  let(:endpoint) { cm.zipped_moab_versions.first.zip_endpoint }
+  let(:logger) { instance_double(Logger) }
+
+  before do
+    allow(Audit::CatalogToArchive).to receive(:logger).and_return(logger)
+  end
+
+  describe '#queue_as' do
+    it 'builds the queue name' do
+      expect(job.queue_name).to eq("part_audit_#{endpoint.endpoint_name}")
+    end
+
+    context 'with a different endpoint' do
+      let(:endpoint) { create(:zip_endpoint, endpoint_name: 'north_foobar') }
+
+      it 'adapts the queue name' do
+        expect(job.queue_name).to eq('part_audit_north_foobar')
+      end
+    end
+  end
+
+  describe '#perform' do
+    let(:zmv1) { cm.zipped_moab_versions.first }
+    let(:zmv2) { cm.zipped_moab_versions.second }
+    let(:results) { job.send(:new_results, cm) }
+
+    it 'only checks parts for one endpoint' do
+      other_ep = create(:zip_endpoint)
+      other_zmv = cm.zipped_moab_versions.find_by(zip_endpoint: other_ep)
+      count = cm.zipped_moab_versions.where(zip_endpoint: endpoint).count
+      expect(job).to receive(:check_child_zip_part_attributes).with(ZippedMoabVersion, AuditResults).exactly(count).times
```

Comment:
what about two slightly more explicit expectations, one for `zmv1` and one for `zmv2`?

Reply:
talked F2F, joe tried the approach i suggested, and it introduced a weird order dependency in the rspec expectations.  we agreed this is good enough as-is.